app/locations/controllers.js

```diff
@@ -0,0 +1,28 @@
+const { find, groupBy, sortBy } = require('lodash')
+
+function _getUserLocations(user) {
+  return groupBy(sortBy(user.locations, 'title'), 'location_type')
+}
+
+function locations(req, res) {
+  const userLocations = _getUserLocations(req.session.user)
+  res.render('locations/views/locations.njk', { userLocations })
+}
+
+function setLocation(req, res) {
+  const locationId = req.params.locationId
+  const location = find(req.session.user.locations, ['id', locationId])
+
+  if (locationId && !location) {
+    throw new Error(`Not authorised to switch to location ${locationId}`)
```

Comment:
This could probably be passed to `next` rather than throwing an error in the middle of this method.

Reply:
How would you like this error shown? Does it need to be in the UI? or is it an operational error?